libs/libc/math/lib_roundl.c

```diff
@@ -34,6 +34,9 @@
 #ifdef CONFIG_HAVE_LONG_DOUBLE
 long double roundl(long double x)
 {
+  if (isinf(x) || isnan(x))
```

Comment:
do we need call isinfl/isnanf version?

Reply:
There doesn't appear to be one and I am not sure it is required. `isinf()` and `isnan()` are defined in `math.h`:
```
#define INFINITY    (1.0/0.0)
#define NAN         (0.0/0.0)
:
:
#define isnan(x)    ((x) != (x))
#define isinf(x)    (((x) == INFINITY) || ((x) == -INFINITY))
#define isfinite(x) (!(isinf(x) || isnan(x)))
```
It would appear using the same definition as `double` is fine.

Reply:
should we add a new one like other libc:
https://man7.org/linux/man-pages/man3/finite.3.html

Reply:
I'm not sure it's needed unlike float, where infinity/nan are not just truncated versions of the double value, they have different values:

#### long double
- infinity: 0x7fff000000000000 0000000000000000 
- nan: 0x7fff800000000000 0000000000000000

#### double
- infinity: 0x7fff000000000000 
- nan: 0x7fff800000000000 

#### float
- infinity: 0x7f800000 
- nan: 0x7fc00000  

If you are using `gcc` you can use `__builtin_infl()`, `__builtin_inf()`, `__builtin_inff()`, `__builtin_nanl("")`, `__builtin_nan("")`, and `__builtin_nanf("")`. 

However, reusing `INFINITY` and `NAN` for long doubles is fine from a value perspective.

Reply:
> If you are using `gcc` you can use `__builtin_infl()`, `__builtin_inf()`, `__builtin_inff()`, `__builtin_nanl("")`, `__builtin_nan("")`, and `__builtin_nanf("")`.
>

common code can't use gcc special macro.
 
> However, reusing `INFINITY` and `NAN` for long doubles is fine from a value perspective.

No, INFINITY for double may be different(e.g. x86 has 80bit long double, 64bit double) from long double. You have to define:
```
#define INFINITY_L  (1.0F/0.0L)
#define NAN_L       (0.0F/0.0L)
```
and use these in isinfl/isnanf.

Reply:
Added those plus macros in subsequent commit.